internal/btf/btf.go

```diff
@@ -543,6 +544,51 @@ func NewHandle(spec *Spec) (*Handle, error) {
 	return &Handle{fd}, nil
 }
 
+// NewHandleFromFD creates a BTF handle from a raw fd.
+//
+// You should not use fd after calling this function.
+func NewHandleromFD(fd int) (*Handle, error) {
+	if fd < 0 {
+		return nil, errors.New("invalid fd")
+	}
+
+	return newHandleFromFD(internal.NewFD(uint32(fd)))
+}
+
+// NewHandleFromID returns the BTF handle for a given id.
+//
+// Returns ErrNotExist, if there is no BTF with the given id.
+// Returns ErrNotSupported if BTF is not supported.
+func NewHandleFromID(id TypeID) (*Handle, error) {
+	if err := haveBTF(); err != nil {
```

Comment:
I could make a function that goes directly from ID to `Spec`, but it seemed likely that once the BTF library is exported that there will be desire for going from a BTF ID to a `Handle` in other use cases as well.

Reply:
It's been a while, but I think I meant the call to `haveBTF` is probably superfluous here.

Reply:
Ah, yes. I think I had that for when it was possible to construct BTF specs from a consumer of the library (when I had all of the type changes) since this could have been the first entry to BTF in the kernel in that case. It is not needed now that all of that is gone. Will remove.

Reply:
Addressed in b87cf4a